File: 2021/02.rb
file = "./02.txt"

data = File.open(file).each_line.map{|l| [l.split(" ").first, l.split(" ").last.to_i]}
location = depth = aim = 0
data.each do |line|
  if line[0] == "forward"
    location += line[1]
    depth += aim * line[1] # Part 2 only
  else
    adjust = line[0] == "down" ? line[1] : -line[1]
    aim += adjust
  end
end

puts location * depth

File: 2021/02.rs

use std::fs::File;
use std::io::{self, BufRead};
use std::path::Path;

const DATA: &str = "./input.txt";

fn main() {
  let instructions = vec!["forward 5", "down 5", "forward 8", "up 3", "down 8", "forward 2"];
  let mut location = 0;
  let mut depth = 0;
  let mut aim = 0;
  for line in instructions.iter() {
    let split_line = line.split(" ");
    let vec = split_line.collect::<Vec<&str>>();
    match vec[0] {
      "forward" => {
        let units: i32 = vec[1].parse().unwrap();
        location += units;
        depth += aim * units;
      }
      "down" => {
        let units: i32 = vec[1].parse().unwrap();
        aim += units;
      }
      // Up
      _ => {
        let units: i32 = vec[1].parse().unwrap();
        aim -= units;
      }
    }
  };
  println!("{}", location * depth)
}